services/service-helpers/validator.js

```diff
@@ -73,6 +140,17 @@ v.arrayOfStrings = wrapCheck(function(value) {
   }
 });
 
+v.arrayOfObjects = wrapCheck(function(value) {
+  if (
+    !Array.isArray(value) ||
+    !value.every(function(x) {
+      return typeof x === 'object';
+    })
+  ) {
+    return 'must be an array of objects';
+  }
+});
```

Comment:
Let's use the `arrayOf` validator you wrote, unless you want to back out of that. We can follow up on this PR with another that replaces `arrayOfStrings` and whatever related functions with `arrayOf`.